backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -196,18 +196,24 @@ export class ExperimentAssignmentService {
     });
 
     if (experiments.length) {
-      const dpExpExists = experiments.filter((exp) => exp.id === experimentId);
+      if (experimentId) {
+        const dpExpExists = experiments.filter((exp) => exp.id === experimentId);
```

Comment:
`dpExpExists` seems like that name of a boolean, rather than of an array that will always have either 0 or 1 elements. (Is there a reason `find()` wasn't used here; I think it would be more efficient)